### PR TITLE
Graph/bugs

### PR DESCRIPTION
**Pull request by orhankislal**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/126/

File: src/ports/postgres/modules/graph/apsp.py_in [hunk 1]
```diff
@@ -72,22 +73,25 @@ def graph_apsp(schema_madlib, vertex_table, vertex_id, edge_table,
         edge_params = extract_keyvalue_params(edge_args, params_types, default_args)
 
         # Prepare the input for recording in the summary table
-        if vertex_id is None:
+        if (vertex_id is None) or (vertex_id == ''):
```
Comment (iyerr3) [inline src/ports/postgres/modules/graph/apsp.py_in]: `if not vertex_id`

File: src/ports/postgres/modules/graph/apsp.py_in [hunk 2]
```diff
@@ -72,22 +73,25 @@ def graph_apsp(schema_madlib, vertex_table, vertex_id, edge_table,
...
             v_st = "NULL"
             vertex_id = "id"
         else:
             v_st = vertex_id
-        if edge_args is None:
+
+        if (edge_args is None) or (edge_args == ''):
```
Comment (iyerr3) [inline src/ports/postgres/modules/graph/apsp.py_in]: `if not edge_args`

File: src/ports/postgres/modules/graph/apsp.py_in [hunk 3]
```diff
@@ -72,22 +73,25 @@ def graph_apsp(schema_madlib, vertex_table, vertex_id, edge_table,
...
             e_st = "NULL"
         else:
             e_st = edge_args
-        if grouping_cols is None:
+
+        if (grouping_cols is None) or (grouping_cols == ''):
```
Comment (iyerr3) [inline src/ports/postgres/modules/graph/apsp.py_in]: `if not grouping_cols`

File: src/ports/postgres/modules/graph/apsp.py_in [hunk 4]
```diff
@@ -163,15 +167,16 @@ def graph_apsp(schema_madlib, vertex_table, vertex_id, edge_table,
         # We keep a summary table to keep track of the parameters used for this
         # APSP run. This table is used in the path finding function to eliminate
         # the need for repetition.
-        plpy.execute(""" CREATE TABLE {out_table}_summary  (
+        summary_table = add_postfix(out_table,"_summary")
```
Comment (iyerr3) [inline src/ports/postgres/modules/graph/apsp.py_in]: space after comma (in multiple places)

File: src/ports/postgres/modules/graph/apsp.py_in [hunk 5]
```diff
@@ -72,22 +73,25 @@ def graph_apsp(schema_madlib, vertex_table, vertex_id, edge_table,
...
-        if grouping_cols is None:
+
+        if (grouping_cols is None) or (grouping_cols == ''):
             g_st = "NULL"
             glist = None
         else:
             g_st = grouping_cols
             glist = split_quoted_delimited_str(grouping_cols)
 
+
```
Comment (iyerr3) [inline src/ports/postgres/modules/graph/apsp.py_in]: Don't need the extra line.

File: src/ports/postgres/modules/graph/bfs.py_in [hunk 6]
```diff
@@ -125,35 +127,39 @@ def graph_bfs(schema_madlib, vertex_table, vertex_id, edge_table,
                                             default_args)
 
         # Prepare the input for recording in the summary table
-        if vertex_id is None:
-            v_st= "NULL"
+        if (vertex_id is None) or (vertex_id == ''):
```
Comment (iyerr3) [inline src/ports/postgres/modules/graph/bfs.py_in]: `if not vertex_id` (similar change in multiple places)

File: src/ports/postgres/modules/graph/bfs.py_in [hunk 7]
```diff
@@ -125,35 +127,39 @@ def graph_bfs(schema_madlib, vertex_table, vertex_id, edge_table,
...
+            v_st = "NULL"
```
Comment (iyerr3) [inline src/ports/postgres/modules/graph/bfs.py_in]: I suggest using `''` to indicate no input. `"NULL"` as a string is different from `NULL`.

File: src/ports/postgres/modules/graph/bfs.py_in [hunk 8]
```diff
@@ -169,7 +175,7 @@ def graph_bfs(schema_madlib, vertex_table, vertex_id, edge_table,
         if grouping_cols is not None and grouping_cols is not '':
```
Comment (iyerr3) [inline src/ports/postgres/modules/graph/bfs.py_in]: `if grouping_cols`

File: src/ports/postgres/modules/graph/pagerank.py_in [hunk 9]
```diff
@@ -84,11 +85,11 @@ def pagerank(schema_madlib, vertex_table, vertex_id, edge_table, edge_args,
         edge_params = extract_keyvalue_params(edge_args, params_types, default_args)
 
         # populate default values for optional params if null
-        if damping_factor is None:
+        if (damping_factor is None) or (damping_factor == ''):
             damping_factor = 0.85
-        if max_iter is None:
+        if (max_iter is None) or (max_iter == ''):
```
Comment (iyerr3) [inline src/ports/postgres/modules/graph/pagerank.py_in]: Why are `max_iter` and `damping_factor` strings?

**Comment by orhankislal** [comment]
Thanks for the comments. I'll fix them right away.

File: src/ports/postgres/modules/graph/pagerank.py_in (hunk 9 above)
Comment (njayaram2) [inline src/ports/postgres/modules/graph/pagerank.py_in]: They are `double` and `int` I guess.

File: src/ports/postgres/modules/graph/pagerank.py_in [hunk 10]
```diff
@@ -84,11 +85,11 @@ def pagerank(schema_madlib, vertex_table, vertex_id, edge_table, edge_args,
...
             max_iter = 100
-        if vertex_id is None:
+        if (vertex_id is None) or (vertex_id == ''):
```
Comment (njayaram2) [inline src/ports/postgres/modules/graph/pagerank.py_in]: `if not vertex_id` should work here too.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/130/

**Comment by orhankislal** [comment]
@iyerr3 Do you have any other comments on this PR?

File: src/ports/postgres/modules/graph/bfs.py_in [hunk 11]
```diff
@@ -47,8 +48,8 @@ def _validate_bfs(vertex_table, vertex_id, edge_table, edge_params,
         out_table,'BFS')
 
     _assert((max_distance >= 0) and isinstance(max_distance,int),
-        """Graph BFS: Invalid max_distance type or value ({0}), must be integer, 
-        be greater than or equal to 0 and be less than max allowable integer 
+        """Graph BFS: Invalid max_distance type or value ({0}), must be integer,
+        be greater than or equal to 0 and be less than max allowable integer
         (2147483647).""".
```
Comment (iyerr3) [inline src/ports/postgres/modules/graph/bfs.py_in]: This can be replaced with 'INT_MAX'

**Comment by iyerr3** [comment]
Looks good to merge.

File: src/ports/postgres/modules/graph/bfs.py_in (hunk 11 above)
Comment (orhankislal) [inline src/ports/postgres/modules/graph/bfs.py_in]: Should we print 'INT_MAX' as text or should we pass INT_MAX as a parameter to `_validate_bfs` to use here?

**Pull request closed without merging**